### ISSUE TITLE
Having 'git' in fileserver_backends and no gitfs_remotes defined causes a crash

### ISSUE THREAD

**Issue by tveastman** [saltstack/salt#37980]
On upgrade to salt `2016.11.0+ds-1` my salt-master wouldn't start, giving the following exception traceback:

```[INFO    ] Setting up the Salt Master
[INFO    ] Preparing the root key for local communication
[INFO    ] Starting up the Salt Master
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
UnboundLocalError: local variable 'repo_obj' referenced before assignment
Traceback (most recent call last):
  File "/usr/bin/salt-master", line 22, in <module>
    salt_master()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 90, in salt_master
    master.start()
  File "/usr/lib/python2.7/dist-packages/salt/cli/daemons.py", line 208, in start
    self.master.start()
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 518, in start
    self._pre_flight()
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 471, in _pre_flight
    fileserver.init()
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/__init__.py", line 496, in init
    self.servers[fstr]()
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/gitfs.py", line 158, in init
    PER_REMOTE_OVERRIDES, PER_REMOTE_ONLY)
  File "/usr/lib/python2.7/dist-packages/salt/utils/gitfs.py", line 2214, in init_remotes
    repo_obj.mountpoint('frap')
UnboundLocalError: local variable 'repo_obj' referenced before assignment
Traceback (most recent call last):
  File "/usr/bin/salt-master", line 22, in <module>
    salt_master()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 90, in salt_master
    master.start()
  File "/usr/lib/python2.7/dist-packages/salt/cli/daemons.py", line 208, in start
    self.master.start()
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 518, in start
    self._pre_flight()
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 471, in _pre_flight
    fileserver.init()
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/__init__.py", line 496, in init
    self.servers[fstr]()
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/gitfs.py", line 158, in init
    PER_REMOTE_OVERRIDES, PER_REMOTE_ONLY)
  File "/usr/lib/python2.7/dist-packages/salt/utils/gitfs.py", line 2214, in init_remotes
    repo_obj.mountpoint('frap')
UnboundLocalError: local variable 'repo_obj' referenced before assignment
```

My git-related configuration looked like:
```
fileserver_backend:
  - roots
  - git

#gitfs_remotes:
#    - https://github.com/saltstack-formulas/docker-formula
```

Note that I had the `git` backend but no `gitfs_remotes` defined.

**My interim solution**: Uncommenting the `gitfs_remotes` stanza seems to fix the problem, as does commenting out the `git` `fileserver_backend`. In my case, since I'm not using it, I simply removed it.

```
Salt Version:
           Salt: 2016.11.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.6
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.6
```

**Comment by p-thurner** [comment]
Also experiencing this behavior on the same salt-master version.

**Comment by Ch3LL** [comment]
@tveastman okay so that was added in this commit: https://github.com/saltstack/salt/pull/33650/commits/ee620af665a832bd352183e5fc61954b1aaff5e2

and it looks like it sets `repo_obj` in the following for loop: `for remote in remotes:` so if you do not have any remotes defined we see tht error because repo_obj is never set.

I'm thinking a solution here is to simpy log an error in as an error but continue starting the master. Any thoughts @terminalmage ? Do you agree with that approach?

**Comment by Ch3LL** [comment]
Also just to clarify @p-thurner you also have git set in your file_roots but no remotes defined correct? just making sure you are running into the same behavior.

**Comment by terminalmage** [comment]
Fixed in #38044, this is debugging code that I was aware made it into my PR for per-saltenv config params, and I thought I had opened a PR to remove it, but I guess I didn't.

Thanks for bringing this to my attention, I'm sorry this caused trouble for you.
